contracts/0.8.9/oracle/BaseOracle.sol

```diff
@@ -0,0 +1,342 @@
+// SPDX-FileCopyrightText: 2023 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.9;
+
+import { SafeCast } from "@openzeppelin/contracts-v4.4/utils/math/SafeCast.sol";
+
+import { UnstructuredStorage } from "../lib/UnstructuredStorage.sol";
+import { Versioned } from "../utils/Versioned.sol";
+import { AccessControlEnumerable } from "../utils/access/AccessControlEnumerable.sol";
+
+import { IReportAsyncProcessor } from "./HashConsensus.sol";
+
+
+interface IConsensusContract {
+    function getIsMember(address addr) external view returns (bool);
+
+    function getCurrentFrame() external view returns (
+        uint256 refSlot,
+        uint256 reportProcessingDeadlineSlot
+    );
+
+    function getChainConfig() external view returns (
+        uint256 slotsPerEpoch,
+        uint256 secondsPerSlot,
+        uint256 genesisTime
+    );
+
+    function getFrameConfig() external view returns (uint256 initialEpoch, uint256 epochsPerFrame);
+}
+
+
+abstract contract BaseOracle is IReportAsyncProcessor, AccessControlEnumerable, Versioned {
+    using UnstructuredStorage for bytes32;
+    using SafeCast for uint256;
+
+    error AddressCannotBeZero();
+    error AddressCannotBeSame();
+    error VersionCannotBeSame();
+    error UnexpectedChainConfig();
+    error OnlyConsensusContractCanSubmitReport();
+    error RefSlotCannotBeLessThanProcessingOne(uint256 refSlot, uint256 processingRefSlot);
+    error RefSlotMustBeGreaterThanProcessingOne(uint256 refSlot, uint256 processingRefSlot);
+    error RefSlotCannotDecrease(uint256 refSlot, uint256 prevRefSlot);
+    error ProcessingDeadlineMissed(uint256 deadline);
+    error UnexpectedRefSlot(uint256 consensusRefSlot, uint256 dataRefSlot);
+    error UnexpectedConsensusVersion(uint256 expectedVersion, uint256 receivedVersion);
+    error UnexpectedDataHash(bytes32 consensusHash, bytes32 receivedHash);
+
+    event ConsensusContractSet(address indexed addr, address indexed prevAddr);
```

Comment:
`ConsensusContractSet` -> `ConsensusHashContractSet` to not get confused

Reply:
done